### ISSUE TITLE
Site config errors disappear immediately

### ISSUE THREAD

**Issue by hopkinscl** [sourcegraph/sourcegraph#37759]
- **Sourcegraph version:** Reproduced on 3.41 and Sourcegraph.com
- **Platform information:** Reproduced on demo.sourcegraph.com, Sourcegraph.com, and self-hosted customer instance (Kubernetes/AWS EKS)

#### Steps to reproduce:

1. Navigate to site config
2. Add an unnecessary character (e.g. a duplicate comma or curly brace) and save changes

**Comment by sourcegraph-bot-2** [comment]
Heads up @sourcegraph/iam - the "team/iam" label was applied to this issue.

**Comment by miveronese** [comment]
I tested this on a local instance and on the scale testing instance and, in both cases, couldn't reproduce the reported bug. The error message was displayed as expected instead of just quickly flashing on the screen.

But I noticed the following behavior:
1. User adds an unnecessary character to the site configuration
2. User sees the error message
3. User discards the changes
4. The error message, instead of disappearing, still shows up. 
5. Only after the page is refreshed, the error message disappears. 

The expected behavior, IMO, would be:
- after the user discards the changes, the error messages disappear.

Will work on a fix for this.
